integration-test/main.go

```diff
@@ -822,6 +842,289 @@ func testDLQFlow() {
+		os.Exit(53)
+	}
+	defer resp.Body.Close()
+	
+	if resp.StatusCode != http.StatusPreconditionFailed {
+		log.Println("Expected 412 for time too close to present, got:", resp.StatusCode)
+		os.Exit(54)
+	}
+	
+	log.Println("Scheduled message error cases test passed")
+}
+
+func testConcurrentScheduledMessages() {
+	log.Println("Testing concurrent scheduled messages")
+	channelID := generalChannelID
+	
+	// Create 5 messages with different future times (all at least 2 minutes in the future)
+	futureTimes := []time.Duration{
+		2*time.Minute + 30*time.Second,
+		3*time.Minute,
+		3*time.Minute + 30*time.Second,
+		4*time.Minute,
+		4*time.Minute + 30*time.Second,
+	}
+	
+	// Schedule messages
+	for i, futureTime := range futureTimes {
+		scheduledTime := time.Now().Add(futureTime).Format(time.RFC3339)
+		
+		req, _ := http.NewRequest(http.MethodPost, brokerBaseURL+"/channel/"+channelID+"/broadcast", strings.NewReader(payload))
+		req.Header.Add(headerContentType, contentType)
+		req.Header.Add(headerChannelToken, token)
+		req.Header.Add(headerProducerID, producerID)
+		req.Header.Add(headerProducerToken, token)
+		req.Header.Add(headerScheduledFor, scheduledTime)
+		
+		resp, err := client.Do(req)
+		if err != nil {
+			log.Println("Error scheduling concurrent message", i, ":", err)
+			os.Exit(55 + i)
+		}
+		defer resp.Body.Close()
+		
+		if resp.StatusCode != http.StatusCreated {
+			body, _ := io.ReadAll(resp.Body)
+			log.Println("Error scheduling concurrent message", i, ":", resp.StatusCode, string(body))
+			os.Exit(60 + i)
+		}
+	}
+	
+	// Set up consumers to receive all messages
+	messagesExpected := len(futureTimes) * pushConsumerCount
+	wg := &sync.WaitGroup{}
+	wg.Add(messagesExpected)
+	
+	for index := 0; index < pushConsumerCount; index++ {
+		consumerHandler[consumerIDPrefix+strconv.Itoa(index)] = func(s string, rw http.ResponseWriter, r *http.Request) {
+			defer func() {
+				if r := recover(); r != nil {
+					log.Println("Recovered", r)
+				}
+			}()
+			body, _ := io.ReadAll(r.Body)
+			if string(body) != payload {
+				consumerAssertionFailed = true
+				log.Println("error - assertion failed for", s)
+			}
+			if r.Header.Get(headerContentType) != contentType {
+				consumerAssertionFailed = true
+				log.Println("error - assertion failed for", s)
+			}
+			rw.WriteHeader(http.StatusNoContent)
+			wg.Done()
+		}
+	}
+	
+	// Wait for all scheduled messages to be delivered
+	timeoutDuration := 6 * time.Minute
+	if waitTimeout(wg, timeoutDuration) {
+		log.Println("Timed out waiting for concurrent scheduled messages delivery")
+		os.Exit(65)
+	}
+	
+	// Check status endpoint for final counts
+	statusURL := brokerBaseURL + "/channel/" + channelID + "/messages-status"
+	req, _ := http.NewRequest(http.MethodGet, statusURL, nil)
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Println("Error getting messages status:", err)
+		os.Exit(66)
+	}
+	defer resp.Body.Close()
+	
+	body, _ := io.ReadAll(resp.Body)
```

Comment:
Assertion needs to be done on unmarshalled `body`? (similar to https://github.com/newscred/webhook-broker/pull/224#discussion_r2096546734)

Reply:
Added assertion below